tools/cpp_build/build_all.sh

```diff
@@ -4,37 +4,65 @@ set -ex
 
 SCRIPTPATH="$( cd "$(dirname "$0")" ; pwd -P )"
 
-INSTALL_PREFIX="$1"
-
-if [[ -z $INSTALL_PREFIX ]]; then
-  INSTALL_PREFIX=$SCRIPTPATH/build/Install
-fi
-
 cd $SCRIPTPATH
 
-PYTORCHPATH="$SCRIPTPATH/../../"
+INSTALL_PREFIX="${1:-$SCRIPTPATH/build/install}"
+PYTORCHPATH="$SCRIPTPATH/../.."
 
 if [ ! -d "$PYTORCHPATH/torch/csrc/autograd/generated" ]; then
   echo "Generated files are not present.\nRun the generators through `python setup.py build` or copy the generated files over."
   exit 1
 fi
 
+NO_CUDA=ON
 if [ -x "$(command -v nvcc)" ]; then
   NO_CUDA=OFF
-else
-  NO_CUDA=ON
 fi
 
 ATEN_BUILDPATH="$SCRIPTPATH/build/aten-build"
 LIBTORCH_BUILDPATH="$SCRIPTPATH/build/libtorch-build"
 
+# Build with Ninja if available. It has much cleaner output.
+GENERATE="Unix Makefiles"
+MAKE=make
+if [ -x "$(command -v ninja)" ]; then
+  GENERATE=Ninja
+  MAKE=ninja
+fi
+
+# Code is developed a lot more than released, so default to Debug.
+BUILD_TYPE=${BUILD_TYPE:-Debug}
+
+# Try to build with as many threads as we have cores, default to 4 if the
+# command fails.
+# https://stackoverflow.com/questions/6481005/how-to-obtain-the-number-of-cpus-cores-in-linux-from-the-command-line
+JOBS="$(grep -c '^processor' /proc/cpuinfo)"
+if [[ $? -ne 0 ]]; then
+  JOBS=4
+fi
+
+echo "Building ATen"
 mkdir -p $ATEN_BUILDPATH && cd $ATEN_BUILDPATH
-cmake -DNO_CUDA:BOOL=$NO_CUDA -DAT_LINK_STYLE:STRING=SHARED -DCMAKE_INSTALL_PREFIX:PATH=$INSTALL_PREFIX -DCMAKE_BUILD_TYPE:STRING=Release $PYTORCHPATH/aten
-make -j4 && make install
+cmake -DNO_CUDA:BOOL=$NO_CUDA \
+      -DAT_LINK_STYLE:STRING=SHARED \
+      -DCMAKE_BUILD_TYPE:STRING=$BUILD_TYPE \
+      -DCMAKE_INSTALL_PREFIX:STRING=$INSTALL_PREFIX \
+      -DCMAKE_INSTALL_MESSAGE=NEVER \
+      -G $GENERATE \
```

Comment:
This should read
```
-G "$GENERATE"
```
otherwise it will fail with `"Unix Makefiles"`, since the value contains a space

Reply:
which is also why CI is currently failing for this PR

Reply:
Thanks